astropy/coordinates/angles/core.py

```diff
@@ -180,9 +180,14 @@ def __new__(cls, angle, unit=None, dtype=np.inexact, copy=True, **kwargs):
                     # Possible conversion to `unit` will be done below.
                     angle = u.Quantity(angle, angle_unit, copy=COPY_IF_NEEDED)
 
-            elif isiterable(angle) and not (
-                isinstance(angle, np.ndarray) and angle.dtype.kind not in "SUVO"
-            ):
+            elif isinstance(angle, np.ndarray):
+                if angle.dtype.kind in "SUVO":
+                    angle = [cls(x, unit, copy=COPY_IF_NEEDED) for x in angle]
+
+            elif hasattr(angle, "__array__"):
```

Comment:
Can we not fall through to `Quantity` here? I.e., could this be
```
elif isiterable(angle) and not hasattr(angle, "__array__"):
```

Reply:
Not sure I understand this suggestion or its motivation: `Quantity` inputs should be handled by the branch just above this one (`isinstance(angle, np.ndarray`), so they never end up here ? (even though that won't be true for Quantity 2.0, which avoids inheritance ?)

For reference I've tried
```patch
diff --git a/astropy/coordinates/angles/core.py b/astropy/coordinates/angles/core.py
index 10e40e76af..a56ef6fbb2 100644
--- a/astropy/coordinates/angles/core.py
+++ b/astropy/coordinates/angles/core.py
@@ -184,11 +184,11 @@ class Angle(SpecificTypeQuantity):
                 if angle.dtype.kind in "SUVO":
                     angle = [cls(x, unit, copy=COPY_IF_NEEDED) for x in angle]
 
-            elif hasattr(angle, "__array__"):
-                angle = np.asarray(angle)
-
             elif isiterable(angle):
-                angle = [cls(x, unit, copy=COPY_IF_NEEDED) for x in angle]
+                if hasattr(angle, "__array__"):
+                    angle = np.asarray(angle)
+                else:
+                    angle = [cls(x, unit, copy=COPY_IF_NEEDED) for x in angle]
```
which passes the test suite but of course that's only a stylistic style change as compared with the current state of the branch.

Reply:
I meant to fall through to let `Quantity.__new__` do the `np.asarray` - but have not really thought this all through, so let my comments not stop just merging this.

Reply:
It is already possible to create `Quantity` instances from `pyarrow` arrays, so the bug is specific to `Angle` and the fix should therefore be limited to `Angle` too.

Reply:
[Quickly, writing from the train from Orlando to New York] Indeed, `Quantity` can deal with it, that's why I was hoping one could find a way simply to *pass on* the `pyarrow` arrays rather than deal with them inside `Angle`, as is the problem in current main (since it deals with it wrongly) and remains the case in this PR.

Reply:
This bug and the fact that `Quantity.__new__()` can parse simple strings but only `Angle.__new__()` can parse lists of simple strings both indicate that the differences between `Angle.__new__()` and `Quantity.__new__()` are larger than they should be. But reducing those differences properly should also lead to `Quantity.__new__()` being able to parse strings, which would be a new feature and therefore something we would not be allowed to backport.

Reply:
Agreed, might be good if `Quantity` could also parse list of str (it can parse simple strings already), but definitely not here!